build_tools/github_actions/runner/config/setup.sh

```diff
@@ -6,106 +6,29 @@
 # See https://llvm.org/LICENSE.txt for license information.
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
-# Sets up a GitHub actions runner VM. Requires custom attributes to indicate
-# configuration settings and uses a proxy service to obtain runner registration
-# tokens (https://github.com/google-github-actions/github-runner-token-proxy).
+# Sets up GitHub actions runner services to start and teardown runner.
+# Registers the runner followed by enabling deregister service then 
+# starts the registered runner to accept self-hosted GitHub workflows.
 
 set -euo pipefail
 
-SCRIPT_DIR="$(dirname -- "$( readlink -f -- "$0"; )")";
+echo "Register the self-hoster runner."
+chmod +x /home/runner/config/register.sh
+/home/runner/config/register.sh
```

Comment:
If you follow my suggestion below. I also remembered about the trick for getting an arbitrary user's home directory.

```suggestion
chmod +x /home/runner/config/register.sh
runuser --user runner ~runner/config/register.sh
```